src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,70 +5,133 @@
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
+	
+	private Integer numerator;
+	private Integer denominator;
 
     public Fractional(Integer numerator, Integer denominator) {
-        // TODO implement to fix tests in FractionalTest1
+    	/*numerator can be 0 but not null*/
+    	if(numerator == null) {
+    		throw new IllegalArgumentException();
+    	} else {
+    		this.numerator = numerator;
+    	}
+    	
+    	/*denominator cant be 0 or null*/
+    	if(denominator == null || denominator == 0) {
+    		throw new IllegalArgumentException();
+    	} else {
+    		this.denominator = denominator;
+    	}
+        
+    }
+    
+    public Fractional invertedFractional(Fractional fractional) {
+    	Fractional returnFractional = new Fractional(
+    		fractional.getDenominator(), fractional.getNumerator());
+    	return returnFractional;	
     }
 
     public Integer getNumerator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return this.numerator;
     }
 
     public Integer getDenominator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return this.denominator;
     }
 
     public Float asFloat() {
-        // TODO implement to fix tests in FractionalTest2
-        return -1f;
+        return  ((float)this.numerator/(float) this.denominator);
     }
 
-    @Override
-    public boolean equals(Object obj) {
-        // TODO implement to fix tests in FractionalTest3
-        return false;
-    }
+	@Override
+	public boolean equals(Object obj) {
+		Fractional fractionalObject;
+		/*!Obj is not a Fractional or is null*/
+		if (!obj.getClass().equals(this.getClass()) || obj == null) {
+			throw new IllegalArgumentException();
+		}
+		fractionalObject = (Fractional) obj;
+		if (fractionalObject.getDenominator() == 0
+				|| fractionalObject.getDenominator() == null
+				|| fractionalObject.getNumerator() == null) {
+			throw new IllegalArgumentException();	
+		}
+		/*Denom & Num are both equal*/
+		if (fractionalObject.getNumerator() == this.getNumerator()
+				&& fractionalObject.getDenominator() 
+				== this.getDenominator()) {
+			return true;
+		}
+		/*Numerator is 0*/
+		if(fractionalObject.getNumerator() == 0 
+				&& this.getNumerator() == 0) {
+			return true;
+		}
+		/*Numerators are equal after multiplication with other
+		 * fractionals denominator*/
+		if(this.getNumerator()*fractionalObject.getDenominator() 
```

Comment:
I think we had this before: 
instead of 
```
if (condition) {
  return true; 
}
return false;
```
you can make your life a little easier and write 
`return condition;`

Reply:
Yes that is something i miss quite often.
This case needs to check for three conditions though.
if (condition) {
 return true;
}
if (condition) {
 return true;
}
if (condition) {
 return true;
}
return false;

I guess i should have chained them with else if() and else().

Reply:
Yes i tend to miss that, 
in this case since there are three linked if(conditions) i think i will stay with the if(condition) syntax.
I guess i should have chained them with else if(), & else though.

Reply:
found a way in the new commit